backend/tests/integration/orcid/test_orcid_domain_models.py

```diff
@@ -265,6 +264,20 @@ def test_push_new_work_already_existing_duplicated_external_identifier_exception
             pusher.push()
         assert self.cache.has_work_content_changed(self.conflicting_inspire_record)
 
+    @mock.patch("inspirehep.orcid.domain_models.OrcidPusher._delete_work")
+    def test_push_new_work_already_existing_and_delete_duplicated_records_different_putcodes(
+        self, mock
+    ):
+        with override_config(
+            ORCID_APP_CREDENTIALS={"consumer_key": "0000-0001-8607-8906"}
+        ):
+            self.add_work(
+                "test_orcid_domain_models_TestOrcidPusherPostNewWork_conflicting_no_recid.xml"
+            )
+            pusher = domain_models.OrcidPusher(self.orcid, self.recid, self.oauth_token)
+            pusher.push()
+            assert mock.called
```

Comment:
why mock this? I would do it like the other tests and check the cassette to make sure the DELETE is sent and the response is ok, instead you should add 
```suggestion
            assert self.cache.has_work_content_changed(self.inspire_record)
```

Reply:
I am not sure if ```self.cache.has_work_content_changed(self.inspire_record)``` will be good there. We are randomly choosing the record which is going to be deleted and those records might change. It is why I preferred to check if one of the records was deleted.

Reply:
at the end of the day, the push should succeed, so the record will have been pushed and the cache will agree with the current version.

Reply:
So it should be ```assert not self.cache.has_work_content_changed(self.inspire_record)``` as the record doesn't change.

Reply:
indeed, I forgot the `not` :facepalm: